add collection pr review checklist (#77661)
ansible/ansible: docs/docsite/rst/community/maintainers_guidelines.rst

How to become a maintainer
--------------------------

A person interested in becoming a maintainer and satisfies the :ref:`requirements<maintainer_requirements>` may either self-nominate or be nominated by another maintainer.

To nominate a candidate, create a GitHub issue in the relevant collection repository. If there is no response, the repository is not actively maintained, or the current maintainers do not have permissions to add the candidate, please create the issue in the `ansible/community <https://github.com/ansible/community>`_ repository.

Communicating as a collection maintainer
-----------------------------------------

>  Maintainers must subscribe to the `"Changes impacting collection contributors and maintainers" GitHub repo <https://github.com/ansible-collections/news-for-maintainers>`_ and the `Bullhorn newsletter <https://github.com/ansible/community/wiki/News#the-bullhorn>`_. If you have something important to announce through the newsletter (for example, recent releases), see the `Bullhorn's wiki page <https://github.com/ansible/community/wiki/News#the-bullhorn>`_ to learn how.

Reviewer: MUST

Reviewer: Should we use MUST more in these docs?

Reviewer 2: uppercase LGTM

>  Maintainers must subscribe to the `"Changes impacting collection contributors and maintainers" GitHub repo <https://github.com/ansible-collections/news-for-maintainers>`_ and the `Bullhorn newsletter <https://github.com/ansible/community/wiki/News#the-bullhorn>`_. If you have something important to announce through the newsletter (for example, recent releases), see the `Bullhorn's wiki page <https://github.com/ansible/community/wiki/News#the-bullhorn>`_ to learn how.

Author: ```suggestion
 Maintainers MUST subscribe to the `"Changes impacting collection contributors and maintainers" GitHub repo <https://github.com/ansible-collections/news-for-maintainers>`_ and the `Bullhorn newsletter <https://github.com/ansible/community/wiki/News#the-bullhorn>`_. If you have something important to announce through the newsletter (for example, recent releases), see the `Bullhorn's wiki page <https://github.com/ansible/community/wiki/News#the-bullhorn>`_ to learn how.
```

* If somebody suggests a good idea, mention it or put a thumbs up.
* After merging, thank the author and reviewers for their time and effort.

.. _review_checklist:

> Reviewing bug reports

Reviewer 2: ```suggestion
Reviewing bug reports
```
The reason why it can be worth having the checklist in a separate file is that all contributors can review issues/PRs, not only maintainers. Maybe we could put links to it in different places like in the contributor path doc, etc.
But i'm not sure, maybe it's OK to have it here.
Would be also nice to hear @felixfontein and @briantist 's thoughts.

----------------------

> When users report bugs, verify the behavior reported. Remember to always be kind with your feedback.

Reviewer 3: ```suggestion
When users report bugs, verify the behavior reported. Remember always to be kind with your feedback.
```

This sounds more formal.

> * [ ] Did the user made a mistake in the example code? We often see user errors reported as bugs.

Reviewer: Is example the right term here? Do we mean the steps to reproduce?

Reviewer 2: yep, it would be clearer, suggested changes, PTAL https://github.com/ansible/ansible/pull/77661#discussion_r860523748

> * [ ] Did the user made a mistake in the example code? We often see user errors reported as bugs.

Reviewer 2: ```suggestion
* [ ] Did the user made a mistake in the code they put in the Steps to reproduce issue's section? We often see user errors reported as bugs.
```

> * [ ] Did the user assume an unexpected behavior? Ensure that the related documentation is clear.

Reviewer: This bugs are useful to help us improve documentation

Reviewer 2: SGTM suggested the change, PTAL https://github.com/ansible/ansible/pull/77661#discussion_r860523050

> * [ ] Did the user assume an unexpected behavior? Ensure that the related documentation is clear.

Reviewer 2: ```suggestion
* [ ] Did the user assume an unexpected behavior? Ensure that the related documentation is clear. If not, the issue is useful to help us improve documentation.
```

* [ ] Is there a minimal reproducer? If not, ask the reporter to reduce the complexity to help pinpoint the issue.
* [ ] Is the issue a consequence of wrong-configured environment?
* [ ] If it seems to be real bug, does the behaviour still exist in the most recent release or the development branch?
* [ ] Reproduce the bug, or if you do not have a suitable infrastructure, ask other contributors to reproduce the bug.

Reviewing suggested changes
---------------------------

> When reviewing PRs, first verify the suggested changes do NOT:

Reviewer 3: ```suggestion
When reviewing PRs, verify the suggested changes first. Do not:
```

* [ ] Unnecessarily break backwards compatibility.
* [ ] Bring more harm than value.
* [ ] Introduce non-idempotent solutions.
* [ ] Duplicate already existing features (inside or outside the collection).
* [ ] Violate the :ref:`Ansible development conventions <module_conventions>`.

Other standards to check for in a PR include:

> * [ ] A pull request does not contain a mix of bugfixes and new features that are not tightly related. If yes, ask the author to split the pull request into separate PRs.

Reviewer: MUST NOT

> * [ ] A pull request does not contain a mix of bugfixes and new features that are not tightly related. If yes, ask the author to split the pull request into separate PRs.

Author: ```suggestion
* [ ] A pull request MUST NOT contain a mix of bugfixes and new features that are not tightly related. If yes, ask the author to split the pull request into separate PRs.
```

* [ ] Modules and plugins from ansible-core use ``ansible.builtin.`` as an FQCN prefix when mentioned.
* [ ] When a new option, module, plugin, or return value is added, the corresponding documentation or return sections use ``version_added:`` containing the *collection* version which they will be first released in.

  * This  is typically the next minor release, sometimes the next major release. For example: if 2.7.5 is the current release, the next minor release will be 2.8.0, and the next major release will be 3.0.0).
* [ ] FQCNs are used for ``extends_documentation_fragment:``, unless the author is referring to doc_fragments from ansible-core.
* [ ] New features have corresponding examples in the :ref:`examples_block`.
* [ ] Return values are documented in the :ref:`return_block`.

Review tests in the PR
----------------------
Review the following if tests are applicable and possible to implement for the changes included in the PR:

* [ ] Where applicable, the pull request has :ref:`testing_integration` and :ref:`testing_units`.

> * [ ] All changes are covered. For example, a bug case or a new option separately and in sensible combinations with other options.

Reviewer 3: ```suggestion
* [ ] All changes are covered. For example, a bug case or a new option separately and insensible combinations with other options.
```

Reviewer 2: @ugwutotheeshoes i think it should be separate (`sensible combinations` is important) ~~or we could use another word, `reasonable` maybe~~ Update: i would leave `sensible`

Reviewer 3: Okay, reasonable could work too.

* [ ] Integration tests cover ``check_mode`` if supported.

> * [ ] Integration tests check an actual state of the system, not only what the module reports. For example, if the module changes a file, check that the file was actually changed by using the ``ansible.builtin.stat`` module..

Reviewer 3: ```suggestion
* [ ] Integration tests check the actual state of the system, not only what the module reports. For example, if the module actually changes a file, check that the file was changed by using the ``ansible.builtin.stat`` module..
```

Reviewer 2: i would leave `actually` here as it can look like the file changed but actually didn't, i.e. the word emphasizes this a bit.

* [ ] Integration tests check return values, if applicable.

Review for merge commits and breaking changes
---------------------------------------------

* [ ] The pull request does not contain merge commits. See the GitHub warnings at the bottom of the pull request. If merge commits are present, ask the author to rebase the pull request branch.

> * [ ] If the pull request contains breaking changes, ask the author and the collection maintainers if it is really needed and there is a way to not introduce breaking changes. If breaking changes are in present, they must only appear in the next major release and **not** in a minor or patch release. The only exception are breaking changes caused by security fixes that are absolutely necessary to fix the security issue.

Reviewer 3: ```suggestion
* [ ] If the pull request contains breaking changes, ask the author and the collection maintainers if it really is needed, and there is a way not to introduce breaking changes. If breaking changes are present, they MUST only appear in the next major release and MUST NOT appear in a minor or patch release. The only exception is breaking changes caused by security fixes that are absolutely necessary to fix the security issue.
```

Reviewer 2: LGTM, though i would leave `really` and `absolutely` to emphasize things as unnecessary breaking changes is usually a really bad thing we absolutely try to avoid:)

Reviewer: Maybe use MUST/MUST NOT
